flutter_local_notifications_windows/src/plugin.cpp

```diff
@@ -0,0 +1,227 @@
+		static_cast<uint32_t>(setting.size() + 1 * sizeof(char))));
+	winrt::check_win32(RegSetValueExA(
+		key.get(),
+		"wnsId",
+		0,
+		REG_SZ,
+		reinterpret_cast<const BYTE*>(wnsId.c_str()),
+		static_cast<uint32_t>(wnsId.size() + 1 * sizeof(char))));
+
+	// now, we register app info to the Registry.
+
+	ss.clear();
+	ss.str(std::string());
+	ss << "Software\\Classes\\AppUserModelId\\" << aumid;
+	const auto appInfoKeyPath = ss.str();
+	RegistryKey appInfoKey;
+
+	// create registry key
+	// HKEY_CURRENT_USER\Software\Classes\AppUserModelId\<aumid>
+	winrt::check_win32(RegCreateKeyExA(
+		HKEY_CURRENT_USER,
+		appInfoKeyPath.c_str(),
+		0,
+		nullptr,
+		0,
+		KEY_WRITE,
+		nullptr,
+		appInfoKey.put(),
+		nullptr));
+
+	winrt::check_win32(RegSetValueExA(
+		appInfoKey.get(),
+		"DisplayName",
+		0,
+		REG_SZ,
+		reinterpret_cast<const BYTE*>(appName.c_str()),
+		static_cast<uint32_t>(appName.size() + 1 * sizeof(char))));
+
+	if (iconPath.has_value()) {
+		const auto v = iconPath.value();
+		winrt::check_win32(RegSetValueExA(
+			appInfoKey.get(),
+			"IconUri",
+			0,
+			REG_SZ,
+			reinterpret_cast<const BYTE*>(v.c_str()),
+			static_cast<uint32_t>(v.size() + 1 * sizeof(char))));
+	}
+
+	// combine guid to class id
+	ss.clear();
+	ss.str(std::string());
+	ss << '{' << guid << '}';
+	const auto clsid = ss.str();
+
+	// register the guid of the notification activation callback
+	winrt::check_win32(RegSetValueExA(
+		appInfoKey.get(),
+		"CustomActivator",
+		0,
+		REG_SZ,
+		reinterpret_cast<const BYTE*>(clsid.c_str()),
+		static_cast<uint32_t>(clsid.size() + 1 * sizeof(char))));
+}
+
+/// Register the notification activation callback factory
+/// and the guid of the callback.
+bool RegisterCallback(const std::string& guid, NativeNotificationCallback callback) {
+	DWORD registration{};
+	winrt::guid rclsid(guid);
+	const auto factory_ref = winrt::make_self<NotificationActivationCallbackFactory>();
+	const auto factory = factory_ref.get();
+	factory->callback = callback;
+	winrt::check_hresult(CoRegisterClassObject(
+		rclsid,
+		factory,
+		CLSCTX_LOCAL_SERVER,
+		REGCLS_MULTIPLEUSE,
+		&registration
+	));
+	return true;
+}
+
+bool NativePlugin::registerApp(
+	const string& aumid,
+	const string& appName,
+	const string& guid,
+	const optional<string>& iconPath,
+	NativeNotificationCallback callback
+) {
+	UpdateRegistry(aumid, appName, guid, iconPath);
+	return RegisterCallback(guid, callback);
+}
+
+std::optional<bool> NativePlugin::checkIdentity() {
+  if (!IsWindows8OrGreater()) return false;
+  uint32_t length = 0;
+  auto error = GetCurrentPackageFullName(&length, nullptr);
+  if (error == APPMODEL_ERROR_NO_PACKAGE) return false;
+  else if (error != ERROR_INSUFFICIENT_BUFFER) return std::nullopt;
+  PWSTR fullName = (PWSTR) malloc(length * sizeof(*fullName));
+  if (fullName == nullptr) return std::nullopt;
+  error = GetCurrentPackageFullName(&length, fullName);
+  if (error != ERROR_SUCCESS) return std::nullopt;
+  free(fullName);
```

Comment:
Use a std::vector. You currently have a memory leak in the case of an error.

Reply:
Addressed - thanks! (I copied almost all the code in this file from previous branches and forks, so the review is appreciated)